anymail/webhooks/mailpace.py

```diff
@@ -0,0 +1,140 @@
+    EventType,
+    RejectReason,
+    inbound,
+    tracking,
+)
+from .base import AnymailBaseWebhookView
+
+
+class MailPaceBaseWebhookView(AnymailBaseWebhookView):
+    """Base view class for MailPace webhooks"""
+
+    esp_name = "MailPace"
+
+    def parse_events(self, request):
+        esp_event = json.loads(request.body.decode("utf-8"))
+        return [self.esp_to_anymail_event(esp_event)]
+
+
+class MailPaceTrackingWebhookView(MailPaceBaseWebhookView):
+    """Handler for MailPace delivery webhooks"""
+
+    webhook_key = None
+
+    def __init__(self, **kwargs):
+        self.webhook_key = get_anymail_setting(
+            "webhook_key",
+            esp_name=self.esp_name,
+            kwargs=kwargs,
+            allow_bare=True,
+            default=None,
+        )
+        self.warn_if_no_basic_auth = self.webhook_key is None
+
+        super().__init__(**kwargs)
+
+    # Used by base class
+    signal = tracking
+
+    event_record_types = {
+        # Map MailPace event RecordType --> Anymail normalized event type
+        "email.queued": EventType.QUEUED,
+        "email.delivered": EventType.DELIVERED,
+        "email.deferred": EventType.DEFERRED,
+        "email.bounced": EventType.BOUNCED,
+        "email.spam": EventType.REJECTED,
+    }
+
+    # MailPace doesn't send a signature for inbound webhooks, yet
+    # When/if MailPace does this, move this to the parent class
+    def validate_request(self, request):
+        if self.webhook_key:
+            try:
+                signature_base64 = request.headers["X-MailPace-Signature"]
+                signature = base64.b64decode(signature_base64)
+            except (KeyError, binascii.Error) as error:
+                raise AnymailWebhookValidationFailure(
+                    "MailPace webhook called with invalid or missing signature"
+                ) from error
+
+            verify_key_base64 = self.webhook_key
+
+            verify_key = VerifyKey(base64.b64decode(verify_key_base64))
+
+            message = request.body
+
+            try:
+                verify_key.verify(message, signature)
+            except (CryptoError, ValueError):
+                raise AnymailWebhookValidationFailure(
+                    "MailPace webhook called with incorrect signature"
+                )
+
+    def esp_to_anymail_event(self, esp_event):
+        event_type = self.event_record_types.get(esp_event["event"], EventType.UNKNOWN)
+        payload = esp_event["payload"]
+
+        reject_reason = (
+            RejectReason.SPAM
+            if event_type == EventType.REJECTED
+            else RejectReason.BOUNCED
+            if event_type == EventType.BOUNCED
+            else None
+        )
+        tags = payload.get("tags", [])
+
+        return AnymailTrackingEvent(
+            event_type=event_type,
+            timestamp=parse_datetime(payload["created_at"]),
+            event_id=payload["id"],
+            message_id=payload["message_id"],
```

Comment:
`message_id` is meant to match the `message_id` returned from the backend's `parse_recipient_status`, so that users can match up webhook events with the original sent message. So I think we want MailPace's "internal email id" here: `message_id=payload["id"]`.